status-network-contracts/src/StakeVault.sol

```diff
@@ -317,6 +324,25 @@ contract StakeVault is IStakeVault, Initializable, OwnableUpgradeable {
         _token.transfer(_destination, _amount);
     }
 
+    /**
+     * @notice Ensures that the new lock until timestamp is valid.
+     * @dev Reverts if the new lock until timestamp is not as expected.
+     * @param _newLockUntil The new lock until timestamp.
+     * @param _seconds The time period to stake for.
+     */
+    function _ensureLockUntilValid(uint256 _newLockUntil, uint256 _seconds) internal {
+        uint256 expectedLockUntil = Math.max(lockUntil, block.timestamp) + _seconds;
+        if (_seconds > 0 && _newLockUntil != expectedLockUntil) {
+            // This should never happen, unless there was a malicious
+            // upgrade of stake manager which tries to lock funds for longer than expected.
+            // Also, there's no case where `newLockUntil < expectedLockUntil`.
+            // It should revert in `StakeManager` already.
+            // We still do a `!=` check to be extra safe, in case a stake manager upgrade
+            // removes all other checks.
+            revert StakeVault__InvalidLockEnd();
+        }
```

Comment:
As mentioned in the commit message, we don't have vault's maxMP available here, nor do I think should we.
However, it's needed by StakeMath._calculateLock().
I decided to just use the one line that is necessary to calculate the new lockUntil, instead of inheriting from StakeMath here and trying to find a way to get access to vault.maxMP.